### ISSUE TITLE
[package] ffmpeg/6.1: Failed to build for Android

### ISSUE THREAD

**Issue by arttet** [conan-io/conan-center-index#23728]
### Description

Failed to build "ffmpeg" for Android

ffmpeg/6.1: ERROR: **GNU assembler not found, install/update gas-preprocessor**

### Package and Environment Details

* Package: **ffmpeg/6.1**
* Operating System: **Android**
* Compiler: **Clang 17**
* Conan version: **conan 2.2.3**
* Python version: **Python 3.12**
* Android NDK: **r26c**

### Conan profile

```text
[settings]
arch=armv8
os=Android
os.api_level=32

compiler=clang
compiler.version=17
compiler.libcxx=c++_static
compiler.cppstd=17

[options]
# Workaround: https://github.com/conan-io/conan-center-index/issues/20511
ffmpeg/*:with_libfdk_aac=False

# Workaround: https://github.com/conan-io/conan-center-index/issues/23718
ffmpeg/*:with_libx264=False

# Workaround: https://github.com/conan-io/conan-center-index/issues/23721
ffmpeg/*:with_libx265=False

# Workaround: https://github.com/conan-io/conan-center-index/issues/23724
ffmpeg/*:with_openh264=False

# Workaround: https://github.com/conan-io/conan-center-index/issues/23726
ffmpeg/*:with_libsvtav1=False

# https://docs.conan.io/2/reference/config_files/global_conf.html
[conf]
tools.cmake.cmaketoolchain:generator=Ninja

[tool_requires]
android-ndk/r26c
```

### Steps to reproduce

[Here](https://github.com/arttet/reproduce-bugs/tree/bug/conan-android-ffmpeg) is a minimal example to reproduce the bug.

### Logs

<details><summary>Click to expand log</summary>

```
-------- Installing package ffmpeg/6.1 (30 of 30) --------
ffmpeg/6.1: Building from source
ffmpeg/6.1: Package ffmpeg/6.1:3e5923495f75965373191ee15eaeec0ff7183f35
ffmpeg/6.1: Copying sources to build folder
ffmpeg/6.1: Building your package in C:\Users\runneradmin\.conan2\p\b\ffmpe678d8ea476b81\b
ffmpeg/6.1: Calling generate()
ffmpeg/6.1: Generators folder: C:\Users\runneradmin\.conan2\p\b\ffmpe678d8ea476b81\b\build-debug\conan
ffmpeg/6.1: Generating aggregated env files
ffmpeg/6.1: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
ffmpeg/6.1: Calling build()
ffmpeg/6.1: apply_conandata_patches(): No patches defined in conandata
ffmpeg/6.1: Calling:
 > "/c/users/runneradmin/.conan2/p/b/ffmpe678d8ea476b81/b/src/configure" --disable-shared --enable-static --prefix=/ --bindir=${prefix}/bin --libdir=${prefix}/lib --pkg-config-flags=--static --disable-doc --enable-cross-compile --enable-asm --disable-shared --enable-static --enable-pic --enable-avdevice --enable-avcodec --enable-avformat --enable-swresample --enable-swscale --enable-postproc --enable-avfilter --enable-bzlib --enable-zlib --enable-lzma --enable-iconv --enable-libopenjpeg --disable-libopenh264 --enable-libvorbis --enable-libopus --disable-libzmq --disable-sdl2 --disable-libx264 --disable-libx265 --enable-libvpx --enable-libmp3lame --disable-libfdk-aac --enable-libwebp --enable-libaom --enable-openssl --disable-alsa --disable-libpulse --disable-vaapi --disable-vdpau --disable-libxcb --disable-libxcb-shm --disable-libxcb-shape --disable-libxcb-xfixes --disable-appkit --disable-avfoundation --disable-coreimage --disable-audiotoolbox --disable-videotoolbox --disable-securetransport --disable-cuda --disable-cuvid --enable-nonfree --enable-gpl --disable-vulkan --disable-libsvtav1 --enable-libdav1d --arch=aarch64 --disable-optimizations --disable-mmx --disable-stripping --enable-debug --nm=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/llvm-nm.exe --ar=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/llvm-ar.exe --as=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/llvm-as.exe --strip=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/llvm-strip.exe --cc=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/aarch64-linux-android32-clang.cmd --cxx=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/aarch64-linux-android32-clang++.cmd --ld=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/ld.exe --ranlib=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/llvm-ranlib.exe --pkg-config=/c/users/runneradmin/.conan2/p/b/pkgco9e3b013bb1017/p/bin/pkgconf.exe --target-os=android "--extra-cflags=-fPIC --sysroot C:/Users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/sysroot -g" "--extra-ldflags=--sysroot C:/Users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/sysroot" 
ffmpeg/6.1: RUN: "/c/users/runneradmin/.conan2/p/b/ffmpe678d8ea476b81/b/src/configure" --disable-shared --enable-static --prefix=/ --bindir=${prefix}/bin --libdir=${prefix}/lib --pkg-config-flags=--static --disable-doc --enable-cross-compile --enable-asm --disable-shared --enable-static --enable-pic --enable-avdevice --enable-avcodec --enable-avformat --enable-swresample --enable-swscale --enable-postproc --enable-avfilter --enable-bzlib --enable-zlib --enable-lzma --enable-iconv --enable-libopenjpeg --disable-libopenh264 --enable-libvorbis --enable-libopus --disable-libzmq --disable-sdl2 --disable-libx264 --disable-libx265 --enable-libvpx --enable-libmp3lame --disable-libfdk-aac --enable-libwebp --enable-libaom --enable-openssl --disable-alsa --disable-libpulse --disable-vaapi --disable-vdpau --disable-libxcb --disable-libxcb-shm --disable-libxcb-shape --disable-libxcb-xfixes --disable-appkit --disable-avfoundation --disable-coreimage --disable-audiotoolbox --disable-videotoolbox --disable-securetransport --disable-cuda --disable-cuvid --enable-nonfree --enable-gpl --disable-vulkan --disable-libsvtav1 --enable-libdav1d --arch=aarch64 --disable-optimizations --disable-mmx --disable-stripping --enable-debug --nm=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/llvm-nm.exe --ar=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/llvm-ar.exe --as=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/llvm-as.exe --strip=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/llvm-strip.exe --cc=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/aarch64-linux-android32-clang.cmd --cxx=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/aarch64-linux-android32-clang++.cmd --ld=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/ld.exe --ranlib=/c/users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/bin/llvm-ranlib.exe --pkg-config=/c/users/runneradmin/.conan2/p/b/pkgco9e3b013bb1017/p/bin/pkgconf.exe --target-os=android "--extra-cflags=-fPIC --sysroot C:/Users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/sysroot -g" "--extra-ldflags=--sysroot C:/Users/runneradmin/.conan2/p/andro936eeaecd9865/p/bin/toolchains/llvm/prebuilt/windows-x86_64/sysroot" 
mkdir: cannot create directory '/dev/shm': Read-only file system

Creating /dev/shm directory failed.
POSIX semaphores and POSIX shared memory will not work

mkdir: cannot create directory '/dev/mqueue': Read-only file system

Creating /dev/mqueue directory failed.
POSIX message queues will not work

GNU assembler not found, install/update gas-preprocessor

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.libera.chat.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.

ffmpeg/6.1: ERROR: 
Package '3e5923495f75965373191ee15eaeec0ff7183f35' build failed
ffmpeg/6.1: WARN: Build folder C:\Users\runneradmin\.conan2\p\b\ffmpe678d8ea476b81\b\build-debug
ERROR: ffmpeg/6.1: Error in build() method, line 710
	autotools.configure()
	ConanException: Error 1 while executing
```

</details>

**Comment by arttet** [comment]
Maybe this [PR](https://github.com/conan-io/conan-center-index/pull/23346) will fix the issue.

**Comment by uilianries** [comment]
Addressed by the PR #24566
